src/layouts/Login/SgidLoginCallbackPage.tsx

```diff
@@ -1,14 +1,127 @@
+import {
+  Flex,
+  Text,
+  Icon,
+  VStack,
+  Spinner,
+  Box,
+  Divider,
+} from "@chakra-ui/react"
+import { Link } from "@opengovsg/design-system-react"
+import { useEffect } from "react"
+import { BiChevronRight } from "react-icons/bi"
 import { useLocation } from "react-router-dom"
 
-import { useSgidLogin } from "hooks/loginHooks"
+import { useSgidLogin, useSgidMultiuserLogin } from "hooks/loginHooks"
+
+import { IsomerLogo } from "assets"
 
 export const SgidLoginCallbackPage = (): JSX.Element => {
   const { search } = useLocation()
   const urlSearchParams = new URLSearchParams(search)
   const params = Object.fromEntries(urlSearchParams.entries())
   const { code } = params
 
-  useSgidLogin({ code })
+  const { data: employments } = useSgidLogin({ code })
+
+  const { mutateAsync: onSelectLogin } = useSgidMultiuserLogin()
+
+  useEffect(() => {
+    if (!employments) return
+    if (employments.length === 1) {
+      window.location.replace("/sites")
+    }
+  }, [employments])
+
+  const shouldDisplayUserSelect = employments && employments.length > 1
+
+  const UserSelect = (): JSX.Element => {
```

Comment:
could we put this outside so its dependencies are clear?

Reply:
Hmm it's dependent on a lot of the things used here though! would have a redefine a lot of it